cpp/src/io/avro/avro_reader_impl.cu

```diff
@@ -185,11 +185,9 @@ table reader::Impl::read(int skip_rows, int num_rows) {
   }
 
   if (md_->total_data_size > 0) {
-    device_buffer<uint8_t> block_data(align_size(md_->total_data_size));
     const auto buffer =
         source_->get_buffer(md_->block_list[0].offset, md_->total_data_size);
-    CUDA_TRY(cudaMemcpyAsync(block_data.data(), buffer->data(), buffer->size(),
-                             cudaMemcpyHostToDevice));
+    rmm::device_buffer block_data(buffer->data(), align_size(buffer->size()));
 
```

Comment:
How does the rmm::device_buffer constructor knows if the input ptr is a device pointer or a host mem pointer ? (or do we assume unified address space ?)

Reply:
It uses `cudaMemcpDefault`. It does rely on UVA, but that has been around since CUDA 4.